src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -1340,15 +1283,65 @@ bool RaveParseMsgPackData(EnvironmentBasePtr penv, KinBodyPtr& ppbody, const std
 
 bool RaveParseMsgPackData(EnvironmentBasePtr penv, RobotBasePtr& pprobot, const std::string& data, const AttributesList& atts, rapidjson::Document::AllocatorType& alloc)
 {
-    rapidjson::Document doc(&alloc);
-    MsgPack::ParseMsgPack(doc, data);
-    JSONReader reader(atts, penv, ".msgpack");
     KinBodyPtr pbody;
-    if( reader.ExtractFirst(doc, pbody, alloc) ) {
+    if (RaveParseMsgPackData(penv, pbody, data, atts, alloc)) {
         pprobot = OPENRAVE_DYNAMIC_POINTER_CAST<RobotBase>(pbody);
         return true;
     }
     return false;
 }
 
+///// Support for PGP-encrypted data
+
+bool RaveParseEncryptedURI(EnvironmentBasePtr penv, const std::string& uri, UpdateFromInfoMode updateMode, const AttributesList& atts, rapidjson::Document::AllocatorType& alloc)
+{
+    JSONReader reader(atts, penv, ".gpg");
+    reader.SetURI(uri);
+    rapidjson::Document rEnvInfo(&alloc);
+#if OPENRAVE_CURL
+    if (reader.IsDownloadingFromRemote()) {
+        reader.OpenRemoteDocument(uri, rEnvInfo);
+    } else
+#endif
+    {
+        std::string fullFilename = ResolveURI(uri, std::string(), reader.GetOpenRAVESchemeAliases());
+        if (fullFilename.size() == 0 ) {
+            return false;
+        }
+        OpenRapidJsonDocument(fullFilename, rEnvInfo);
```

Comment:
I'm afraid but using OpenRapidJsonDocument in RaveParseEncryptedURI is wrong.

Reply:
This is now implemented